src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,204 @@
+package duke;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Contains methods that deal with storing and loading Tasks from memory.
+ */
+public class Storage {
+    private String filePath;
+    private String tempFilePath;
+
+    /**
+     * Constructor for creating Storage.
+     *
+     * @param filePath The filePath to the file that stores Tasks in a .txt file.
+     * @param tempFilePath The filePath for a temporary file for use when main file has to be rewritten.
+     */
+    public Storage(String filePath, String tempFilePath) {
+        this.filePath = filePath;
+        this.tempFilePath = tempFilePath;
+    }
+
+    /**
+     * Creates files for storage of Tasks if they do not exist. If they exist, returns list of Tasks.
+     *
+     * @return An ArrayList of Tasks added previously.
+     */
+    public ArrayList<Task> load() {
+        // Ensure file exists
+        File hardDiskTasks = new File(filePath);
+        File tempTasks = new File(tempFilePath);
+        try {
+            hardDiskTasks.createNewFile();
+            tempTasks.createNewFile();
+        } catch (IOException | SecurityException e) {
+            System.out.println("     " + e.getMessage());
+        }
+
+        // Add disk info to taskList
+        ArrayList<Task> pastTasks = loadTasksFromDisk(hardDiskTasks);
+        return pastTasks;
+    }
+
+    /**
+     * Loads Tasks previously added from memory and adds into an ArrayList.
```

Comment:
Lots of JavaDocs and detailed comments for easier understanding which is very helpful!